packages/backend/src/tokens/authenticateContext.ts

```diff
@@ -39,44 +47,76 @@ class AuthenticateContext {
     this.handshakeToken =
-      this.clerkRequest.clerkUrl.searchParams.get(constants.QueryParameters.Handshake) ||
-      this.clerkRequest.cookies.get(constants.Cookies.Handshake);
+      this.getQueryParam(constants.QueryParameters.Handshake) ||
+      this.getCookieWithVariants(constants.Cookies.Handshake);
   }
 
   private initHeaderValues() {
-    const get = (name: string) => this.clerkRequest.headers.get(name) || undefined;
-    this.sessionTokenInHeader = this.stripAuthorizationHeader(get(constants.Headers.Authorization));
-    this.origin = get(constants.Headers.Origin);
-    this.host = get(constants.Headers.Host);
-    this.forwardedHost = get(constants.Headers.ForwardedHost);
-    this.forwardedProto = get(constants.Headers.CloudFrontForwardedProto) || get(constants.Headers.ForwardedProto);
-    this.referrer = get(constants.Headers.Referrer);
-    this.userAgent = get(constants.Headers.UserAgent);
-    this.secFetchDest = get(constants.Headers.SecFetchDest);
-    this.accept = get(constants.Headers.Accept);
+    this.sessionTokenInHeader = this.stripAuthorizationHeader(this.getHeader(constants.Headers.Authorization));
+    this.origin = this.getHeader(constants.Headers.Origin);
+    this.host = this.getHeader(constants.Headers.Host);
+    this.forwardedHost = this.getHeader(constants.Headers.ForwardedHost);
+    this.forwardedProto =
+      this.getHeader(constants.Headers.CloudFrontForwardedProto) || this.getHeader(constants.Headers.ForwardedProto);
+    this.referrer = this.getHeader(constants.Headers.Referrer);
+    this.userAgent = this.getHeader(constants.Headers.UserAgent);
+    this.secFetchDest = this.getHeader(constants.Headers.SecFetchDest);
+    this.accept = this.getHeader(constants.Headers.Accept);
   }
 
   private initCookieValues() {
-    const get = (name: string) => this.clerkRequest.cookies.get(name) || undefined;
-    this.sessionTokenInCookie = get(constants.Cookies.Session);
-    this.clientUat = Number.parseInt(get(constants.Cookies.ClientUat) || '') || 0;
+    this.sessionTokenInCookie = this.getCookieWithVariants(constants.Cookies.Session);
+    this.clientUat = Number.parseInt(this.getCookieWithVariants(constants.Cookies.ClientUat) || '') || 0;
   }
 
   private stripAuthorizationHeader(authValue: string | undefined | null): string | undefined {
     return authValue?.replace('Bearer ', '');
   }
+
+  private getQueryParam(name: string) {
+    return this.clerkRequest.clerkUrl.searchParams.get(name);
+  }
+
+  private getHeader(name: string) {
+    return this.clerkRequest.headers.get(name) || undefined;
+  }
+
+  private getCookieWithVariants(cookieName: string) {
+    const pipeline = new Pipeline<string | undefined, Map<string, string>>()
+      .addOperation(new ReadSuffixedCookieOperation(cookieName, this.publishableKey))
+      .addOperation(new ReadCookieOperation(cookieName));
+    // get first result
+    return pipeline.invoke(this.clerkRequest.cookies).find(Boolean);
```

Comment:
🤔 Instead of introducing this `Pipeline` primitive, can we create some sort of cookies abstraction that handles dealing with the suffix?

This isn't a pattern we have elsewhere and I'm finding myself having to jump to the Pipeline definition to understand what's happening here. If we want to keep this abstraction, some comments might help explaining how it works.